developers/weaviate/client-libraries/python/async.md

```diff
@@ -0,0 +1,196 @@
+
+- `async_client.collections.get("<COLLECTION_NAME>")`: Create a Python object to interact with an existing collection (this does not create a collection)
+- `async_client.is_connected()`: Check the last known connection status to the Weaviate server
+
+## Context manager
+
+The async client can be used in an asynchronous context manager, in a pattern similar to:
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START AsyncContextManager"
+  endMarker="# END AsyncContextManager"
+  language="py"
+/>
+
+When using the async client in a context manager, you do not need to call `.connect()` or `.close()` explicitly. The client handles the connection and disconnection automatically.
+
+## Async usage examples
+
+The async client object largely provides the same functionality as the [synchronous Python client](./index.md), with some key differences. First, the async client is designed to be used in an `async` function or in an [`asyncio` event loop](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio-event-loop). Accordingly, many of the client methods are `async` functions that return [`Coroutines` objects](https://docs.python.org/3/library/asyncio-task.html#coroutine).
+
+To execute an async client method, you must `await` the method call or use `asyncio.run()` to run the method in an event loop.
+
+### Data insertion
+
+In this example, we create a new collection and insert a list of objects into the collection using the async client.
+
+Note the use of a context manager in the async function. The context manager is used to ensure that the client is connected to the server during the data insertion operation.
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START AsyncInsertionExample"
+  endMarker="# END AsyncInsertionExample"
+  language="py"
+/>
+
+### Search & RAG
+
+In this example, we perform retrieval augmented generation (RAG) with hybrid search results using the async client.
+
+Note the use of a context manager in the async function. The context manager is used to ensure that the client is connected to the server during the data insertion operation.
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START AsyncSearchExample"
+  endMarker="# END AsyncSearchExample"
+  language="py"
+/>
+
+### Bulk data insertion
+
+For server-side batch operations, we recommend using the synchronous client and its [`batch` operations](./index.md#batch-imports). The batch operations are designed to handle large amounts of data efficiently through concurrent requests already.
+
+The async client still offers `insert` and `insert_many` methods for data insertion, which can be used in an async context.
+
+### Application-level example
+
+A common use case for the async client is in web applications, where multiple requests are handled concurrently. Here is an indicative, minimal example integrating the async client with [FastAPI](https://fastapi.tiangolo.com/), a popular web framework:
```

Comment:
WDYT about:

- a popular web framework for creating modular web API microservices

(moar buzzwords 😛)

Reply:
Great synergy!